### DISCUSSION TITLE
Add notable contributions threshold

### DISCUSSION THREAD

**Discussion by Milo123459** [lowlighter/metrics#225] [Ideas]
The way it works will allow you to add a threshold between 3 categories: watchers, stargazers or forks.
for example, if you have contributed to desktop/desktop that has  2000 stars and your threshold is 500 stars, it'd display that, so you could do:
plugin_notable_threshold: stars 500, forks 500, watchers 500
Then if the repository meats those conditions it will display!

**Comment by lowlighter** [comment]
Yes it would be a good feature to implement!

Stars, watchers and forks can be retrieved by patching this query:
https://github.com/lowlighter/metrics/blob/master/source/plugins/notable/queries/contributions.graphql

Then patching this condition to add the threshold requirements:
https://github.com/lowlighter/metrics/blob/b7c4f6e5bbc3506a2f0aa59879eb024b103d7850/source/plugins/notable/index.mjs

As for the treshold syntax, maybe we could base it on GitHub's search syntax (which is similar to what you proposed), like `stars:>500 forks:100..200` so we have a "normalized way" to do it 🤔 ?

___

One thing to note though, is that this plugin was originally thought to mimick the GitHub activity section (but only counting orgs where you commited at least once) so that's why it only display the org name and not the repo name.

But maybe we could add a new option to let user display either just org names (in this case we would display org name only if user has contributed to at least **one** repo with matching criterias) or org names + repo name (where only repo matching criterias will be displayed)

**Reply by Milo123459** [reply]
Well, what I meant is that it goes through each repo you have contributed to on the organization then see if the terms are et, and if they are, display the org.
Thanks for the reply tho!

**Reply by lowlighter** [reply]
In this case just consider the first part of the reply 🙂

**Comment by lowlighter** [comment]
Implemented in #240

You can now use:
```yaml
- uses: lowlighter/metrics@latest
  with:
    # ... other options
    plugin_notable: yes
    plugin_notable_filter: stars:>500  
```

Thanks a lot for your suggestion 🙂 !
